src/main/java/org/openrefine/extensions/commons/importer/RelatedCategoryFetcher.java

```diff
@@ -41,34 +52,56 @@ public List<String> getRelatedCategories(String filename, String pageID) throws
         relatedCategories = jsonNode.path("query").path("pages").path(pageId).path("categories");
         toCategoriesColumn = new ArrayList<>();
         for (JsonNode category: relatedCategories) {
-            toCategoriesColumn.add(category.findValue("title").toString());
+            toCategoriesColumn.add(category.findValue("title").asText());
         }
 
         return toCategoriesColumn;
     }
 
+    /*
+     * Returns {@code true} if the iteration has more elements for
+     * which to fetch related categories.
+     * (In other words, returns {@code true} if {@link #next} would
+     * return an element rather than throwing an exception.)
+     *
+     * @return {@code true} if the iteration has more elements
+     */
     @Override
     public boolean hasNext() {
-        // TODO Auto-generated method stub
-        return false;
+
+        return iteratorFileRecords.hasNext();
     }
 
+    /*
+     * This method iterates over each of the categories related to a file
+     * and stores them as a list in the relatedCategories parameter of
+     * each file record
+     *
+     * @return an instance of the FileRecord updated to include its related categories
+     */
     @Override
     public FileRecord next() {
 
-        FileRecord fileRecord;
+        FileRecord fileRecordOriginal;
+        FileRecord fileRecordNew = null;
         if (iteratorFileRecords.hasNext() != false) {
-            fileRecord = iteratorFileRecords.next();
+            fileRecordOriginal = iteratorFileRecords.next();
             try {
-                new FileRecord(fileRecord.fileName, fileRecord.pageId, getRelatedCategories(fileRecord.fileName, fileRecord.pageId));
+                fileRecordNew = new FileRecord(fileRecordOriginal.fileName, fileRecordOriginal.pageId,
+                        getRelatedCategories(fileRecordOriginal.fileName, fileRecordOriginal.pageId));
             } catch (IOException e) {
                 // TODO Auto-generated catch block
                 e.printStackTrace();
```

Comment:
We need to think about what to do in this case. What do you think should happen if we fail to retrieve the categories of one of the files in the list?